CMakeLists.txt

```diff
@@ -75,7 +75,7 @@ add_definitions(-fPIC)
 # A grep of 'std::auto_ptr<' in the header files show 7 hits, and 13 hits in cpp files
 #
 IF (${DISTRO} STREQUAL "Ubuntu_17.04")
-  set (CMAKE_CXX_FLAGS "-Wall -Wno-unknown-pragmas -D_LARGEFILE64_SOURCE -D_GNU_SOURCE -fno-var-tracking-assignments")
+  set (CMAKE_CXX_FLAGS "-Wall -Wno-unknown-pragmas -D_LARGEFILE64_SOURCE -D_GNU_SOURCE -Werror -Wno-deprecated-declarations -fno-var-tracking-assignments")
```

Comment:
This fixes the problem in Ubuntu about not seeing warnings. Still can't capture the warnings about deprecated, but that's a minor thing.

NTC

Reply:
In fact, this should better solved in master branch, then progressed to feature/ngsiv2_registrations. On the one hand, it is not really related with the issue this PR is solved. On the other hand, it may pass a long time until feature/ngsiv2_registrations lands into master.

As a side comment, maybe a comment explaining why the new flag has been added would be fine.  When weird problem occur, that kind of information could become useful.

Reply:
Lets not ...
That "side comment" might be "before we had no warnings at all in Ubuntu, now we have".
It would be to put a comment saying that "before this didn't work, now it does!", only .. where? 
The change is for the entire code base ...

NTC

Reply:
I understand NTC refers to the idea of adding a comment. However, the idea of doing a direct one-liner PR to master with this change stills, doesn't it?

Reply:
Sure, why not.

Reply:
Done in another PR

NTC